REGRESSION(276012@main): CMake fails with GObject-Introspection disabled

aperezdc opened the pull request:
#### 3a1c08120188470f531a1a3a4837b496cc900bd7
<pre>
REGRESSION(276012@main): CMake fails with GObject-Introspection disabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=269377">https://bugs.webkit.org/show_bug.cgi?id=269377</a>

Reviewed by Michael Catanzaro and Philippe Normand.

* Source/WebKit/PlatformGTK.cmake: Quote the expansion of ${GI_VERSION}
  to ensure VERSION_GREATER_EQUAL has at least an empty string as value
  to compare against; otherwise when the variable is undefined there was
  no left-hand side of the comparison, which resulted in CMake erroring
  due to wrong syntax.

Canonical link: <a href="https://commits.webkit.org/276019@main">https://commits.webkit.org/276019@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6a761f23b7717564dc2137e69be2f80159537a1c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43555 "Passed style check") | [💥 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22591 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [💥 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45970 "An unexpected error occured. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46194 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39684 "Built successfully") 
| | [💥 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26407 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20007 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/46194 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44129 "Passed tests") | [💥 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/26407 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 api-mac](https://ews-build.webkit.org/#/builders/14/builds/45970 "An unexpected error occured. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/46194 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| | [💥 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/26407 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/14/builds/45970 "An unexpected error occured. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1611 "Built successfully") | 
| | [💥 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/26407 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/14/builds/45970 "An unexpected error occured. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47740 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| | [💥 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18588 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/51/builds/20007 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/47740 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| | [💥 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20011 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/14/builds/45970 "An unexpected error occured. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [💥 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/47740 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| | [💥 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20189 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5930 "Built successfully and passed tests") | [💥 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19639 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/PlatformGTK.cmake:
```diff
@@ -369,7 +369,7 @@ if (ENABLE_MEDIA_STREAM)
 endif ()
 
-if (${GI_VERSION} VERSION_GREATER_EQUAL 1.79.2)
+if ("${GI_VERSION}" VERSION_GREATER_EQUAL 1.79.2)
```
Maybe it would be less confusing to just check `if (ENABLE_INTROSPECTION AND GI_VERSION VERSION_GREATER_EQUAL 1.79.2)`

Unless I'm mistaken, we shouldn't even need to defereference `${}` the `GI_VERSION` here, right? I wonder why I did that. Maybe it really is needed?

aperezdc replied:
CMake doesn't do short-circuit evaluation, so we would need instead:

```cmake
if (ENABLE_INTROSPECTION)
    if (${GI_VERSION} VERSION_GREATER_EQUAL 1.79.2)
        # ...
```

IMO that would be a little bit more cumbersome than expanding in between quotes and knowing that CMake expands undefined variables in strings into an empty value.

aperezdc committed:
REGRESSION(276012@main): CMake fails with GObject-Introspection disabled
https://bugs.webkit.org/show_bug.cgi?id=269377

Reviewed by Michael Catanzaro and Philippe Normand.

* Source/WebKit/PlatformGTK.cmake: Quote the expansion of ${GI_VERSION}
  to ensure VERSION_GREATER_EQUAL has at least an empty string as value
  to compare against; otherwise when the variable is undefined there was
  no left-hand side of the comparison, which resulted in CMake erroring
  due to wrong syntax.

Canonical link: https://commits.webkit.org/276019@main
